metomi/rose/loc_handlers/rsync.py

```diff
@@ -24,6 +26,62 @@
 from metomi.rose.popen import RosePopenError
 
 
+class PreRsyncCheckError(Exception):
+    """Error to raise if we assume the loc is for use with rsync,
+    but attempting to use it as an rsync loc fails."""
+    BASE_MESSAGE = (
+        'Rose tried all other file install handlers and'
+        ' decided this must be an Rsync handler.\n\t'
+    )
+
+    def __init__(self, dict_, cmd=None):
+        for key, value in dict_.items():
+            if isinstance(value, TextIOWrapper):
+                setattr(self, key, value.read())
+            else:
+                setattr(self, key, value)
+
+        # Convert command into something the debugger can try:
+        try:
+            self.cmd = ' '.join(cmd)
+        except TypeError:
+            self.cmd = cmd
+
+        # Handle ``test -e nonexistant`` where no useful error is
+        # provided:
+        if (
+            self.returncode == 1
+            and self.stderr == ''
+            and self.stdout == ''
+        ):
+            self.stderr = f'File "{cmd[-1]}" does not exist.'
+
+        if self.returncode == 255:
+            host = dict_['args'][dict_['args'].index('-n') + 1]
+            self.message = (
+                self.BASE_MESSAGE
+                + 'If it is then host'
+                f' "{host}"'
+                ' is uncontactable (ssh 255 error).'
+            )
+        else:
+            self.message = (
+                self.BASE_MESSAGE
+                + f'`{self.cmd}` failed with:'
+                + indent(
+                    f'\nreturncode: {self.returncode}'
+                    f'\nstdout:     {self.stdout}'
+                    f'\nstderr:     {self.stderr}',
+                    prefix='    ',
+                )
+            )
+        super().__init__(self.message)
```

Comment:
Why this `super.__init__`?

Reply:
Mistake - it doesn't need to be any more

```suggestion
```

Reply:
I thought that there might be a better way - will mod.